### DISCUSSION TITLE
All my devices are offline

### DISCUSSION THREAD

**Discussion by ghost** [Ylianst/MeshCentral#6284] [Issues]
Hi all,
I have a strange problem.
I was on holiday for 2 weeks and I shutdown all the devices including the router during holiday.
When I booted the router back I saw that the external IP was changed so I logged in the server locally and via my domain.
The devices are offline when logging locally or via Domain.
Is there a way to solve this?
Thanks.

**Comment by si458** [comment]
What is your config.json like?

**Reply by ghost** [reply]
I've changed the Cert line to the new IP


{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
"__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "77.20.140.86",
    "selfUpdate": false
  },
  "domains": {
    "": {
	"title": "",
	"_title2": "SimonD",
	"_welcomeText": "Test",
	"_loginPicture":"LoginPic.JPG",
	"welcomePicture": "WelcomePic.JPG",
	"welcomePictureFullScreen": true,
        "mstsc": true,
	"userSessionIdleTimeout": 15
    }
  }
}

**Reply by si458** [reply]
You didn't remember from what I said last time.
DONT USE AN IP ADDRESS!!!
Because u have now changed ur ip address in the cert value, all ur remote agent devices won't connect anymore!
U must use a dns name!
Buy a cheap domain, create an A record and point it to ur external ip address.
Or I'd u want a free dns name, use something like duckdns.org and get a free ddns which includes scripts/software that will keep the ip address updated if ur external ip address changes

**Reply by ghost** [reply]
No, I always had an IP and it worked.
I changed the IP in the Cert because the external IP changed.
But the devices are offline even if I connect locally to the server and not via domain.
Anyway I changed the config to the domain and still everything is offline.
Something in the network changed and I don't what.

**Reply by si458** [reply]
Yes because u changed the cert value!
Every time u change the cert value, certificates get recreated and existing agent can then no longer connect!
So u have to uninstall the agents and redownload/reinstall them
This is why we tell people to use DNS!
then when ur ip address changes, it doesn't matter,
Just change the ip address of where ur dns points too and everything carry on working...

**Reply by ghost** [reply]
Ok anyway I inserted the the DNS in the config and updated the ip address in the DNS settings and still not working.

**Reply by silversword411** [reply]
> I inserted the the DNS

Too late. Your existing agents are broken and will never reconnect unless you can get your server back on the old IP.

**Reply by ghost** [reply]
So let me get this straight.
If I'm changing the external IP or Domain then basically I'm screwed and need to reconnect all the devices manually.
Is there a migration process? What happens if I'm changing a domain and I have 100 devices?`
Another question: How can I connect to the server locally without a domain?

**Reply by si458** [reply]
> So let me get this straight.
> If I'm changing the external IP or Domain then basically I'm screwed and need to reconnect all the devices manually.
> Is there a migration process? What happens if I'm changing a domain and I have 100 devices?`
> Another question: How can I connect to the server locally without a domain?

Yes sadly u are stuffed, this is why we tell people to use a DNS name and NOT an ip address, then u only have change the ip address of where the dns points to and u can still get control again!

There is a new feature I added the other month, in the console called `msh`, which will let u change a few things in the msh file, like the url of ur server,, but it's not guaranteed to work and the remote device MUST BE ONLINE!
but It was mainly added to allow edit of the customised image/name, etc

And u can only access meshcentral locally without a dns if ur in LAN mode and all ur computers are OK the same network range as ur meshcentral server
